File: wx-login-service/src/main/java/com/example/wxlogin/entity/GpuKnowledge.java
package com.example.wxlogin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * GPU显卡维修知识库实体
 */
@Data
@TableName("gpu_knowledge")
public class GpuKnowledge {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 问题分类 */
    private String category;

    /** 问题描述 */
    private String question;

    /** 常见原因 */
    private String causes;

    /** 排查方法 */
    private String diagnosis;

    /** 维修方案 */
    private String solution;

    /** 难度评估 */
    private String difficulty;

    /** 维修成本 */
    private String cost;

    /** 成功率 */
    private String successRate;

    /** 排序序号 */
    private Integer sortOrder;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}

File: gpu_diagnostic_app/lib/core/models/gpu_knowledge.dart
/// GPU 知识库条目模型
class GpuKnowledge {
  final int id;
  final String category;
  final String question;
  final String causes;
  final String diagnosis;
  final String solution;
  final String? difficulty;
  final String? cost;
  final String? successRate;

  GpuKnowledge({
    required this.id,
    required this.category,
    required this.question,
    required this.causes,
    required this.diagnosis,
    required this.solution,
    this.difficulty,
    this.cost,
    this.successRate,
  });

  factory GpuKnowledge.fromJson(Map<String, dynamic> json) {
    return GpuKnowledge(
      id: json['id'] ?? 0,
      category: json['category'] ?? '',
      question: json['question'] ?? '',
      causes: json['causes'] ?? '',
      diagnosis: json['diagnosis'] ?? '',
      solution: json['solution'] ?? '',
      difficulty: json['difficulty'],
      cost: json['cost'],
      successRate: json['successRate'] ?? json['success_rate'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'category': category,
      'question': question,
      'causes': causes,
      'diagnosis': diagnosis,
      'solution': solution,
      'difficulty': difficulty,
      'cost': cost,
      'successRate': successRate,
    };
  }
}
